runtime/testy/runtime.py

```diff
@@ -91,17 +91,18 @@ def nonTestModePrint(*args):
           controlState = "auto"
           continue
         elif newBadThing.event == BAD_EVENTS.ENTER_IDLE and controlState != "idle":
+          controlState = "idle"
           break
         print(newBadThing.event)
         nonTestModePrint(newBadThing.data)
         if newBadThing.event in restartEvents:
+          restartCount += 1
           if (not emergency_stopped and newBadThing.event is BAD_EVENTS.EMERGENCY_STOP):
             emergency_stopped = True #somehow kill student code using other method? right now just restarting on e-stop
           break
```

Comment:
Given that you added line 94 `controlState = "idle"`, I think you should also add it here and remove it from line 105.